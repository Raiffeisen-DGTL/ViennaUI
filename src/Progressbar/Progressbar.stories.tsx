import React from 'react';
import { Story, Meta } from 'storybook';
import { Progressbar, ProgressbarProps } from './Progressbar';

export default {
    title: 'Development/Progressbar',
    component: Progressbar,
} as Meta;

export const Overview: Story<ProgressbarProps> = (args) => {
    return <Progressbar {...args} />;
};
