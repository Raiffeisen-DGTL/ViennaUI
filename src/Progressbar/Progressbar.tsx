import React, { FC } from 'react';
import {
    Box,
    PropsBox,
    CircleSVG,
    CircleProgress,
    Line,
    Progress,
    CircleBack,
    CircleBox,
    CircleContent,
} from './Progressbar.styles';
import { Loading } from './Loading';
import { BoxStyled } from '../Utils/styled';

export type ProgressSize = 's' | 'm' | 'l';
export type ProgressColor =
    | 'moscow100'
    | 'osaka100'
    | 'accent'
    | 'geneva100'
    | 'oslo120'
    | 'seattle140'
    | 'oslo100'
    | 'miami100'
    | 'sochi100'
    | 'paris100'
    | 'tokyo100'
    | 'dubai100'
    | 'nice100';

export interface ProgressbarProps extends BoxStyled<typeof Box, PropsBox> {
    view?: 'line' | 'circle';
    size?: ProgressSize;
    color?: ProgressColor;
    value?: number;
    /** Shows loading state, if view is 'line' */
    loading?: boolean;
    indeterminate?: boolean;
    width?: number;
}

export const Progressbar: FC<ProgressbarProps> = ({
    size = 'm',
    view = 'line',
    color = 'oslo120',
    value,
    children,
    indeterminate,
    loading,
    ...attrs
}) => {
    switch (view) {
        case 'circle': {
            return (
                <CircleBox {...attrs}>
                    <CircleSVG $size={size}>
                        <CircleBack $size={size} />
                        <CircleProgress $size={size} $color={color} $indeterminate={indeterminate} $value={value} />
                    </CircleSVG>
                    {children && <CircleContent $size={size}>{children}</CircleContent>}
                </CircleBox>
            );
        }
        case 'line':
        default: {
            return (
                <Box {...attrs} $size={size}>
                    <Line />
                    <Progress $indeterminate={indeterminate} $width={value && value > 100 ? 100 : value} $color={color}>
                        {!indeterminate && loading && <Loading />}
                    </Progress>
                </Box>
            );
        }
    }
};

Progressbar.displayName = 'Progressbar';
