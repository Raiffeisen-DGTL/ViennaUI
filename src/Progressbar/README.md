# Progressbar

Компонент индикации прогресса информирует пользователя о состоянии текущих процессов в интерфейсе.

Может быть использован для:

-   Отображения статуса загрузки страницы, процесса или конкретного файла в интерфейсе
-   Отображения прогресса пользователя по выполнению задачи. В данном случае рекомендуем рассмотреть применение компонента Stepper, он может лучше подойти для такой задачи.

## Импорт

```
import { Progressbar } from 'vienna-ui';
```

## Свойства / Props

| Prop | Type | Default | Description |
| --- | --- | --- | --- |
| view | "line" \| "circle" \| undefined | "line" | Тип отображения: прямая или окружность |
| size | "s" \| "m" \| "l" \| undefined | "m" |
| color | "accent" \| "moscow100" \| "osaka100" \| "geneva100" \| "oslo120" \| "seattle140" \| "oslo100" \| "miami100" \| "sochi100" \| "paris100" \| "tokyo100" \| "dubai100" \| "nice100" \| undefined | "oslo120" |
| value\* | number | 0 | Значение загрузки |
| loading | boolean \| undefined | false | Отображается анимация на заполненной полосе компонента. Доступно только для view='line' |

## Использование

```
<Progressbar value={34} />
```

## Использование с доченими компонентами

В Progressbar со значением `view` равным `circle`, можно передать контент внутрь компонента

```
<Progressbar view='circle' size='l' value={34}>
    <div style={{ height: '100%', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>34%</div>
</Progressbar>
<Progressbar view='circle' size='l' value={34}>
    <Image />
</Progressbar>
```

## Режим отображения

##### Свойство `view`

Progressbar имеет 2 режима отображения `view`: `line` (линейный - по умолчанию) и `circle` (круглый)

```
<Progressbar value={34} />
<Progressbar value={76} view='circle' />
```

## Размеры

##### Свойство `size`

Доступны 3 размера для каждого из дизайнов: `s`, `m` (по умолчанию), `l`

## Дизайн

##### Свойство `design`

Доступны следующие значения свойства `color`: `moscow100`, `osaka100`, `accent`, `geneva100`, `oslo120`, `seattle140`, `oslo100`, `miami100`, `sochi100`, `paris100`, `tokyo100`, `dubai100`, `nice100`.

```
<Progressbar color='moscow100' value={34} />
<Progressbar color='moscow100' view='circle' value={34} />
```

## Режим загрузки

##### Свойство `loading`

Устанавливается с помощью свойства `loading`. Доступно только для `view='line'`

```
<Progressbar loading value={64} />
```
