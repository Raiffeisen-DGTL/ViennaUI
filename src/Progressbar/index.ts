export { Progressbar, type ProgressbarProps } from './Progressbar';
