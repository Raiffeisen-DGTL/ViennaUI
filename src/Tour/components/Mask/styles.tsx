import styled, { css } from 'styled-components';
import { color } from 'vienna.tokens';

export const Wrapper = styled.div<{ $isOverlayHidden: boolean }>`
    left: 0;
    top: 0;
    position: fixed;
    z-index: 99999;
    pointer-events: none;
    color: ${color.primary.brand.primary};
    ${({ $isOverlayHidden }) =>
        $isOverlayHidden
            ? css`
                  opacity: 0;
              `
            : css`
                  opacity: 0.24;
              `};
`;

export const SvgWrapper = styled.svg<{
    $windowWidth: number;
    $windowHeight: number;
    $wpt: number | string;
    $wpl: number | string;
}>`
    position: fixed;
    ${({ $windowHeight, $windowWidth, $wpl, $wpt }) => css`
        width: ${$windowWidth}px;
        height: ${$windowHeight}px;
        left: ${Number($wpl)}px;
        top: ${Number($wpt)}px;
    `};
`;

export const MaskArea = styled.rect<{ $x: number; $y: number; $width: number; $height: number; $rx: number }>`
    fill: black;
    ${({ $height, $width, $x, $y, $rx = 16 }) => css`
        x: ${$x}px;
        y: ${$y}px;
        width: ${$width}px;
        height: ${$height}px;
        rx: ${$rx};
    `};
`;

export const MaskRect = styled.rect<{ $windowWidth: number; $windowHeight: number; $maskId?: string }>`
    fill: currentColor;
    ${({ $windowHeight, $windowWidth, $maskId }) => {
        return css`
            x: 0px;
            y: 0px;
            width: ${$windowWidth}px;
            height: ${$windowHeight}px;
            mask: url(#${$maskId});
        `;
    }};
`;

export const HighlightedArea = styled.rect<{
    $x: number;
    $y: number;
    $width: number;
    $height: number;
}>`
    pointer-events: auto;
    fill: transparent;
    display: none;
    ${({ $x, $y, $width, $height }) => css`
        x: ${$x}px;
        y: ${$y}px;
        width: ${$width}px;
        height: ${$height}px;
    `};
`;

export const ClickArea = styled.rect<{
    $width: number;
    $height: number;
    $top: number;
    $left: number;
    $windowWidth: number;
    $windowHeight: number;
    $clipId?: string;
}>`
    fill: currentColor;
    pointer-events: auto;
    ${({ $clipId, $windowHeight, $windowWidth }) => css`
        x: 0px;
        y: 0px;
        width: ${$windowWidth}px;
        height: ${$windowHeight}px;
        clip-path: url(#${$clipId});
    `};
`;
