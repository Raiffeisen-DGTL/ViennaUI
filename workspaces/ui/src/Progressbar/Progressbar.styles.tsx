import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const line = getPresets('progressbar.line', {
    size: 'size',
    back: null,
});

const lineProgress = getPresets('progressbar.line.progress', {
    base: null,
    color: 'color',
});

const circle = getPresets('progressbar.circle', {
    size: 'size',
    back: null,
});

const circleBox = getPresets('progressbar.circle.box', {
    size: 'size',
});

const circleProgress = getPresets('progressbar.circle.progress', {
    base: null,
    color: 'color',
});

const circleContent = getPresets('progressbar.circle.content', {
    size: 'size',
});

export const Box = styled.div<any>`
    position: relative;
    width: 100%;

    ${line.size}
`;

export const Line = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;

    ${line.back}
`;

export const Progress = styled.div<any>`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;

    ${({ width }) =>
        width !== undefined &&
        css`
            width: ${width}%;
        `}

    ${lineProgress.base}
    ${lineProgress.color}
`;

export const CircleSVG = styled.svg<any>`
    fill: none;

    ${circleBox.size}
`;

const RADIUS = {
    s: 32,
    m: 44,
    l: 64,
};

const calculateCircleLength = (r) => Math.PI * 2 * r;

const calculateProgressLength = (r, value) => {
    const circleLength = calculateCircleLength(r);
    return (circleLength * (100 - value)) / 100;
};

export const CircleBack = styled.circle<any>`
    ${circle.size}
    ${circle.back}
`;

export const CircleProgress = styled.circle<any>`
    stroke-linecap: round;
    transform: rotate(-90deg);
    transform-origin: 50%;

    ${circle.size}
    ${circleProgress.base}
    ${circleProgress.color}

    ${({ size, value }) => css`
        stroke-dasharray: ${(calculateCircleLength(RADIUS[size]) * value) / 100}
            ${calculateProgressLength(RADIUS[size], value)};
    `}
`;

export const CircleBox = styled.div`
    position: relative;
    display: inline-flex;
`;

export const CircleContent = styled.div<any>`
    position: absolute;
    border-radius: 50%;
    overflow: hidden;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);

    ${circleContent.size}
`;
