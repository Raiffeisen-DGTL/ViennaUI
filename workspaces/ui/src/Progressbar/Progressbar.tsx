import React from 'react';
import {
    Box,
    CircleSVG,
    CircleProgress,
    Line,
    Progress,
    CircleBack,
    CircleBox,
    CircleContent,
} from './Progressbar.styles';
import { Loading } from './Loading';

interface Props {
    view?: 'line' | 'circle';
    size?: 's' | 'm' | 'l';
    color?:
        | 'moscow100'
        | 'osaka100'
        | 'accent'
        | 'geneva100'
        | 'oslo120'
        | 'seattle140'
        | 'oslo100'
        | 'miami100'
        | 'sochi100'
        | 'paris100'
        | 'tokyo100'
        | 'dubai100'
        | 'nice100';
    // design is depricated property
    design?: 'red' | 'orange' | 'yellow' | 'green' | 'darkBlue' | 'darkGray' | 'lightBlue';
    value: number;
    /** Shows loading state, if view is 'line' */
    loading?: boolean;
}

interface HTMLAttributeProps {
    id?: string;
    className?: string;
    role?: string;
    spellCheck?: boolean;
    tabIndex?: number;
}

export type ProgressbarProps = Props & HTMLAttributeProps;

export const Progressbar: React.FC<ProgressbarProps> = (props: React.PropsWithChildren<ProgressbarProps>) => {
    const { size, view, color, design, value, children, loading, ...attrs } = props;

    // Маппинг старых значений цвета на новые. Удалить по окончанию поддержки старых значений цветов
    const designToColorMapping = {
        red: 'moscow100',
        orange: 'osaka100',
        yellow: 'accent',
        green: 'geneva100',
        darkBlue: 'oslo120',
        darkGray: 'seattle140',
        lightBlue: 'oslo100',
    };

    // Функция для маппинга старых значений цвета на новые. Удалить по окончанию поддержки старых значений цветов
    const getColor = () => {
        if (design) {
            // eslint-disable-next-line no-console
            console.warn(
                `Свойство design устарело. В следующих версиях ДС оно будет удалено. Пожалуйста, используйте вместо него свойство color.`
            );

            return designToColorMapping[design];
        }

        return color;
    };

    switch (view) {
        case 'circle': {
            return (
                <CircleBox {...attrs}>
                    <CircleSVG size={size}>
                        <CircleBack size={size} />
                        {/* Для того чтобы удалить поддержку старых цветов нужно заменить color={getColor()} на color={color} */}
                        <CircleProgress size={size} color={getColor()} value={value} />
                    </CircleSVG>
                    {children && <CircleContent size={size}>{children}</CircleContent>}
                </CircleBox>
            );
        }
        case 'line':
        default: {
            return (
                <Box {...attrs} size={size}>
                    <Line />
                    {/* Для того чтобы удалить поддержку старых цветов нужно заменить color={getColor()} на color={color} */}
                    <Progress width={value > 100 ? 100 : value} color={getColor()}>
                        {loading && <Loading />}
                    </Progress>
                </Box>
            );
        }
    }
};

Progressbar.displayName = 'Progressbar';
Progressbar.defaultProps = {
    view: 'line',
    size: 'm',
    color: 'oslo120',
};

export default Progressbar;
