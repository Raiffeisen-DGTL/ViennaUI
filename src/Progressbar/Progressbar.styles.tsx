import styled, { css, keyframes } from 'styled-components';
import { progressbar } from 'vienna.ui-theme';
import { getPresets } from '../Utils/styling';
import { ProgressColor, ProgressSize } from './Progressbar';

const line = getPresets(
    progressbar.line,
    'progressbar.line'
)({
    size: '$size',
    back: null,
});

const lineProgress = getPresets(
    progressbar.line.progress,
    'progressbar.line.progress'
)({
    base: null,
    color: '$color',
});

const circle = getPresets(
    progressbar.circle,
    'progressbar.circle'
)({
    size: '$size',
    back: null,
});

const circleBox = getPresets(
    progressbar.circle.box,
    'progressbar.circle.box'
)({
    size: '$size',
});

const circleProgress = getPresets(
    progressbar.circle.progress,
    'progressbar.circle.progress'
)({
    base: null,
    color: '$color',
});

const circleContent = getPresets(
    progressbar.circle.content,
    'progressbar.circle.content'
)({
    size: '$size',
});

const indeterminateLineAnimation = keyframes`
    0% {
        left: 0;
        width: 1%;
    }
    25% {
        left: 25%;
        width: 50%;
    }
    50% {
        left: 99%;
        width: 1%;
    }
    75% {
        left: 25%;
        width: 50%;
    }
    100% {
        left: 0;
        width: 1%;
    }
`;

const indeterminateCircleAnimation = keyframes`
    0% {
        stroke-dasharray: 10% 280%;
        transform: rotate(-90deg);
    }
    25% {
        stroke-dasharray: 100% 280%;
        transform: rotate(45deg);
    }
    50% {
        stroke-dasharray: 10% 280%;
        transform: rotate(270deg);
    }
    75% {
        stroke-dasharray: 100% 280%;
        transform: rotate(45deg);
    }
    100% {
        stroke-dasharray: 10% 280%;
        transform: rotate(-90deg);
    }
`;

export interface PropsBox {
    $size?: ProgressSize;
}
export const Box = styled.div<PropsBox>`
    position: relative;
    width: 100%;

    ${line.size}
`;

export const Line = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;

    ${line.back}
`;

interface PropsProgress {
    $indeterminate?: boolean;
    $width?: number;
    $color?: ProgressColor;
}
export const Progress = styled.div<PropsProgress>`
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;

    ${({ $width }) =>
        $width !== undefined &&
        css`
            width: ${$width}%;
        `}

    ${({ $indeterminate }) =>
        $indeterminate &&
        css`
            animation: ${indeterminateLineAnimation} linear 1.2s infinite;
        `}

    ${lineProgress.base}
    ${lineProgress.color}
`;

interface PropsCircleSVG {
    $size?: ProgressSize;
}
export const CircleSVG = styled.svg<PropsCircleSVG>`
    fill: none;

    ${circleBox.size}
`;

const RADIUS = {
    s: 32,
    m: 44,
    l: 64,
};

const calculateCircleLength = (r: number) => Math.PI * 2 * r;

const calculateProgressLength = (r: number, value: number) => {
    const circleLength = calculateCircleLength(r);
    return (circleLength * (100 - value)) / 100;
};

interface PropsCircleBack {
    $size?: ProgressSize;
}
export const CircleBack = styled.circle<PropsCircleBack>`
    ${circle.size}
    ${circle.back}
`;

interface PropsCircleProgress {
    $size?: ProgressSize;
    $color: ProgressColor;
    $value?: number;
    $indeterminate?: boolean;
}
export const CircleProgress = styled.circle<PropsCircleProgress>`
    stroke-linecap: round;
    transform: rotate(-90deg);
    transform-origin: 50%;

    ${circle.size}
    ${circleProgress.base}
    ${circleProgress.color}

    ${({ $size, $value }) => css`
        stroke-dasharray: ${$size && $value && (calculateCircleLength(RADIUS[$size]) * $value) / 100}
            ${$size && $value && calculateProgressLength(RADIUS[$size], $value)};
    `}

    ${({ $indeterminate }) =>
        $indeterminate &&
        css`
            animation: ${indeterminateCircleAnimation} linear 1.2s infinite;
        `}
`;

export const CircleBox = styled.div`
    position: relative;
    display: inline-flex;
`;

interface PropsCircleContent {
    $size?: ProgressSize;
}
export const CircleContent = styled.div<PropsCircleContent>`
    position: absolute;
    border-radius: 50%;
    overflow: hidden;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);

    ${circleContent.size}
`;
