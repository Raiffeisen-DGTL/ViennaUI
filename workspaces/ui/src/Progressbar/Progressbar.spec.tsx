import React from 'react';
import Progressbar, { ProgressbarProps } from './Progressbar';
import { Progress } from './Progressbar.styles';

test('Progressbar', () => {
    const snap = snapshot.render(<Progressbar value={50} />);
    expect(snap).toMatchSnapshot();
});

test('Progressbar w/ view', () => {
    const views: ProgressbarProps['view'][] = ['line', 'circle'];

    views.forEach((view) => {
        const snap = snapshot.render(<Progressbar value={50} view={view} />);
        expect(snap).toMatchSnapshot(view);
    });
});

test('Progressbar w/ size', () => {
    const sizes: ProgressbarProps['size'][] = ['s', 'm', 'l'];

    sizes.forEach((size) => {
        const snap = snapshot.render(<Progressbar value={50} size={size} />);
        expect(snap).toMatchSnapshot(size);
    });
});

test('Progressbar w/ color', () => {
    const colors: ProgressbarProps['color'][] = [
        'moscow100',
        'osaka100',
        'accent',
        'geneva100',
        'oslo120',
        'seattle140',
        'oslo100',
        'miami100',
        'sochi100',
        'paris100',
        'tokyo100',
        'dubai100',
        'nice100',
    ];

    colors.forEach((color) => {
        const snap = snapshot.render(<Progressbar value={50} color={color} />);

        expect(snap).toMatchSnapshot(color);
    });
});

test('Progressbar w/ view w/ child', () => {
    const sizes: ProgressbarProps['size'][] = ['s', 'm', 'l'];

    sizes.forEach((size) => {
        const snap = snapshot.render(
            <Progressbar view='circle' size={size} value={50}>
                <div style={{ height: '100%', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
                    34%
                </div>
            </Progressbar>
        );
        expect(snap).toMatchSnapshot(size);
    });
});

test('Progressbar w/ loading', () => {
    const snap = snapshot.render(<Progressbar value={50} loading />);
    expect(snap).toMatchSnapshot();
});
