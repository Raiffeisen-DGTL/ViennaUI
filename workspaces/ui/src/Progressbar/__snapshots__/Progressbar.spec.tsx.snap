// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progressbar 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 gXNIna"
    color="oslo120"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: accent 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 cIlUC"
    color="accent"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: dubai100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 kZlsvb"
    color="dubai100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: geneva100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 bcESoA"
    color="geneva100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: miami100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 flCHzJ"
    color="miami100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: moscow100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 idCTTt"
    color="moscow100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: nice100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 btyVOd"
    color="nice100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: osaka100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 gwWMHq"
    color="osaka100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: oslo100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 jGYoUl"
    color="oslo100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: oslo120 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 gXNIna"
    color="oslo120"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: paris100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 jmweAI"
    color="paris100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: seattle140 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 iXKiRB"
    color="seattle140"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: sochi100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 cPdhhf"
    color="sochi100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ color: tokyo100 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 kUmUku"
    color="tokyo100"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ loading 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 gXNIna"
    color="oslo120"
    width="50"
  >
    <svg
      class="Loadingstyles__Box-sc-1vysjtz-1 dCOwmU"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <pattern
          height="20"
          id="strips"
          patternTransform="rotate(-30)"
          patternUnits="userSpaceOnUse"
          width="80"
          x="10"
          y="10"
        >
          <rect
            class="Loadingstyles__PatternRect-sc-1vysjtz-0 hLkmys"
            height="10"
            width="80"
            x="0"
            y="0"
          />
        </pattern>
      </defs>
      <rect
        fill="url(#strips)"
        height="100%"
        width="100%"
        x="0"
        y="0"
      />
    </svg>
  </div>
</div>
`;

exports[`Progressbar w/ size: l 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 dUtaQS"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 gXNIna"
    color="oslo120"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ size: m 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 gXNIna"
    color="oslo120"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ size: s 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 ebQjGZ"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 gXNIna"
    color="oslo120"
    width="50"
  />
</div>
`;

exports[`Progressbar w/ view w/ child: l 1`] = `
<div
  class="Progressbarstyles__CircleBox-sc-9u8n53-6 ezdLOV"
>
  <svg
    class="Progressbarstyles__CircleSVG-sc-9u8n53-3 kcaFuf"
  >
    <circle
      class="Progressbarstyles__CircleBack-sc-9u8n53-4 dztelO"
    />
    <circle
      class="Progressbarstyles__CircleProgress-sc-9u8n53-5 dBoxtR"
      color="oslo120"
      value="50"
    />
  </svg>
  <div
    class="Progressbarstyles__CircleContent-sc-9u8n53-7 jhFrJS"
  >
    <div
      style="height:100%;display:flex;align-items:center;justify-content:center"
    >
      34%
    </div>
  </div>
</div>
`;

exports[`Progressbar w/ view w/ child: m 1`] = `
<div
  class="Progressbarstyles__CircleBox-sc-9u8n53-6 ezdLOV"
>
  <svg
    class="Progressbarstyles__CircleSVG-sc-9u8n53-3 bxpMcD"
  >
    <circle
      class="Progressbarstyles__CircleBack-sc-9u8n53-4 cIIcZb"
    />
    <circle
      class="Progressbarstyles__CircleProgress-sc-9u8n53-5 hefzLA"
      color="oslo120"
      value="50"
    />
  </svg>
  <div
    class="Progressbarstyles__CircleContent-sc-9u8n53-7 cWJLkG"
  >
    <div
      style="height:100%;display:flex;align-items:center;justify-content:center"
    >
      34%
    </div>
  </div>
</div>
`;

exports[`Progressbar w/ view w/ child: s 1`] = `
<div
  class="Progressbarstyles__CircleBox-sc-9u8n53-6 ezdLOV"
>
  <svg
    class="Progressbarstyles__CircleSVG-sc-9u8n53-3 gldwKv"
  >
    <circle
      class="Progressbarstyles__CircleBack-sc-9u8n53-4 fTwEea"
    />
    <circle
      class="Progressbarstyles__CircleProgress-sc-9u8n53-5 fCjBiZ"
      color="oslo120"
      value="50"
    />
  </svg>
  <div
    class="Progressbarstyles__CircleContent-sc-9u8n53-7 jClxxC"
  >
    <div
      style="height:100%;display:flex;align-items:center;justify-content:center"
    >
      34%
    </div>
  </div>
</div>
`;

exports[`Progressbar w/ view: circle 1`] = `
<div
  class="Progressbarstyles__CircleBox-sc-9u8n53-6 ezdLOV"
>
  <svg
    class="Progressbarstyles__CircleSVG-sc-9u8n53-3 bxpMcD"
  >
    <circle
      class="Progressbarstyles__CircleBack-sc-9u8n53-4 cIIcZb"
    />
    <circle
      class="Progressbarstyles__CircleProgress-sc-9u8n53-5 hefzLA"
      color="oslo120"
      value="50"
    />
  </svg>
</div>
`;

exports[`Progressbar w/ view: line 1`] = `
<div
  class="Progressbarstyles__Box-sc-9u8n53-0 fayEqV"
>
  <div
    class="Progressbarstyles__Line-sc-9u8n53-1 lfqWLM"
  />
  <div
    class="Progressbarstyles__Progress-sc-9u8n53-2 gXNIna"
    color="oslo120"
    width="50"
  />
</div>
`;
